api/models/release.go

```diff
@@ -671,3 +668,93 @@ func waitForTemplate(bucket string, id string) error {
 	fmt.Printf("ns=kernel at=waitForTemplate.tick error=unknown bucket=%q release=%q\n", bucket, id)
 	return fmt.Errorf("unknown error")
 }
+
+// monitorReleasePromotion observes a release's rollout by verifying the app's CF stack status
+// Sends a notifcation on success or failure. This function blocks.
+// TODO: this should be in provider.ReleasePromote()
+func waitForPromotion(r *Release) {
```

Comment:
We definitely could use CF SNS events. We went with this method (knowing polling could fail) as the quickest way of getting promotion events.

Another option is lambda acting on CF events via SNS. One problem is finding a reliable way of knowing a stack update was a promotion or some other stack change (e.g scaling a process) without or little state management.